### PR TITLE
Fixes non-constant expressions in switch-block if-statements

### PR DESCRIPTION
**Pull request by Hinaichigo**
It seems that there's a [bug](http://www.byond.com/forum/post/2750423) in the BYOND compiler that allows non-constant expressions in switch-block if-statements of the format `if(x to y)` to be compiled, but they are treated as `null`. So I went through all of the cases of that that I could find and changed them.
[bugfix]

### CODE REVIEW

File: code/game/objects/structures/bedsheet_bin.dm [hunk 1]
```diff
@@ -151,12 +151,12 @@ LINEN BINS
 
 
 /obj/structure/bedsheetbin/update_icon()
-	switch(amount)
-		if(0)
-			icon_state = "linenbin-empty"
-		if(1 to amount / 2)
-			icon_state = "linenbin-half"
-		else				icon_state = "linenbin-full"
```
Comment (Exxion) [inline code/game/objects/structures/bedsheet_bin.dm:159]: >if `amount` is between 1 and half of `amount`

Yeah this one would really make you think even without the bug

Comment (Hinaichigo) [reply code/game/objects/structures/bedsheet_bin.dm:159]: Yeeah.. lol.

Comment (Hinaichigo) [reply code/game/objects/structures/bedsheet_bin.dm:159]: I don't really like using initial() but I'm not sure of a better way here besides making a new variable.

Comment (Exxion) [reply code/game/objects/structures/bedsheet_bin.dm:159]: I feel the same way about `initial()`. Making a new variable would be the only alternative, though. I don't really think anyone cares much either way about this particular object.

Comment (DamianX) [reply code/game/objects/structures/bedsheet_bin.dm:159]: If only someone had made a system that provided all objects with a `maxHealth` var.

Comment (Exxion) [reply code/game/objects/structures/bedsheet_bin.dm:159]: Are you saying this thing should treat its number of contained bedsheets as health?

Comment (DamianX) [reply code/game/objects/structures/bedsheet_bin.dm:159]: I was reading the other changes and didn't think too hard about this one. Yeah, this one is fine as it is, I guess.

**Pull request merged**